java/src/test/java/ai/rapids/cudf/ColumnVectorTest.java

```diff
@@ -3936,4 +3936,88 @@ void testMakeList() {
       assertColumnsAreEqual(expected, created);
     }
   }
+
+  @Test
+  void testReplaceLeafNodeInList() {
+    try (
+        ColumnVector c1 = ColumnVector.fromInts(1, 2);
+        ColumnVector c2 = ColumnVector.fromInts(8, 3);
+        ColumnVector c3 = ColumnVector.fromInts(9, 8);
+        ColumnVector c4 = ColumnVector.fromInts(2, 6);
+        ColumnVector expected = ColumnVector.makeList(c1, c2, c3, c4);
+         ColumnVector child1 = ColumnVector.decimalFromDoubles(DType.create(DType.DTypeEnum.DECIMAL64, 3), RoundingMode.HALF_UP, 770.892, 961.110);
+         ColumnVector child2 = ColumnVector.decimalFromDoubles(DType.create(DType.DTypeEnum.DECIMAL64, 3), RoundingMode.HALF_UP, 524.982, 479.946);
+         ColumnVector child3 = ColumnVector.decimalFromDoubles(DType.create(DType.DTypeEnum.DECIMAL64, 3), RoundingMode.HALF_UP, 346.997, 479.946);
+         ColumnVector child4 = ColumnVector.decimalFromDoubles(DType.create(DType.DTypeEnum.DECIMAL64, 3), RoundingMode.HALF_UP, 87.764, 414.239);
+         ColumnVector created = ColumnVector.makeList(child1, child2, child3, child4);
+         ColumnVector newChild = ColumnVector.fromInts(1, 8, 9, 2, 2, 3, 8, 6);
```

Comment:
Wow! This is a valid operation? Conversion between `List<double>` and `List<int>`? Shouldn't we be more restrictive?

Reply:
The whole point for adding this method is to be able to swap out the leaf node. Its not a conversion but a complete rewrite, and its a view that will not change the original vector.